src/utils/DateUtils.js

```diff
@@ -261,63 +287,59 @@ const utils = {
     const day = this.getDate(date)
 
     const matches = {
-      dd: `0${day}`.slice(-2),
       d: day,
-      yyyy: year,
-      yy: String(year).slice(2),
-      MMMM: this.getMonthName(this.getMonth(date), translation.months),
-      MMM: this.getMonthNameAbbr(this.getMonth(date), translation.monthsAbbr),
-      MM: `0${month}`.slice(-2),
-      M: month,
-      o: this.getNthSuffix(this.getDate(date)),
+      dd: `0${day}`.slice(-2),
       E: this.getDayNameAbbr(date, translation.days),
+      o: this.getNthSuffix(this.getDate(date)),
+      M: month,
+      MM: `0${month}`.slice(-2),
+      MMM: this.getMonthNameAbbr(this.getMonth(date), translation.monthsAbbr),
+      MMMM: this.getMonthName(this.getMonth(date), translation.months),
+      yy: String(year).slice(2),
+      yyyy: year,
```

Comment:
Shouldn't this new sorting break everything?
Because the sorting defines which part is replaced first. For example now if you have something like `MMMM` the first thing that is replaced is a single `M` which is used four times on the format instead the long version `MMMM` which is replaced last